Formula/nim.rb

```diff
@@ -14,28 +14,23 @@ class Nim < Formula
 
   head do
     url "https://github.com/nim-lang/Nim.git", branch: "devel"
-    resource "csources" do
-      url "https://github.com/nim-lang/csources.git"
-    end
   end
 
   depends_on "help2man" => :build
 
   def install
     if build.head?
-      resource("csources").stage do
-        system "/bin/sh", "build.sh"
-        (buildpath/"bin").install "bin/nim"
-      end
+      system "/bin/sh", "build_all.sh"
```

Comment:
Does this download anything? Why do we no longer need the `csources` resource?

Reply:
the details are handled inside build_all.sh which handles cloning a csources repo at the correct revision (it's not csources anymore, it's some csources_v1 due to recent changes to avoid breaking previous builds relying on HEAD of csources)

Reply:
I see. In that case, we can't use `build_all.sh` outside of `HEAD`, because formulae should not be downloading anything that isn't declared in a `resource` block at install time. See https://docs.brew.sh/Acceptable-Formulae#we-dont-like-install-scripts-that-download-unversioned-things.

Ideally, we shouldn't be doing this for `HEAD` either, but the existing `resource` block for `csound` isn't checksummed anyway.

Reply:
> I see. In that case, we can't use build_all.sh outside of HEAD

i've updated comments accordingly; build_all.sh is only used for HEAD (in future we can declare a resource for csources_v1, clone it, and then call: `. ci/funs.sh && nimBuildCsourcesIfNeeded` to install it at the right revision; but it should be deferred to future work IMO since this is only done for HEAD)